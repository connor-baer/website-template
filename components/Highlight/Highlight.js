import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { hideVisually } from 'polished';

/**
 * NOTE: Accessibility improvements are taken from
 * http://adrianroselli.com/2017/12/tweaking-text-level-styles.html
 */

const baseStyles = ({ theme }) => css`
  display: inline;
  color: #000;
  background-color: ${theme.colors.y100};

  &::before,
  &::after {
    ${hideVisually()};
  }

  &::before {
    content: ' [highlight start] ';
  }

  &::after {
    content: ' [highlight end] ';
  }

  @media screen and (-ms-high-contrast: active) {
    color: HighlightText;
    background-color: Highlight;
  }

  @media print {
    border: 1pt dotted ${theme.colors.black};
  }
`;

const Highlight = styled('mark')(baseStyles);

export default Highlight;
