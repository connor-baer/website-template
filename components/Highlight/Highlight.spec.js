import React from 'react';

import Highlight from './Highlight';

describe('Highlight', () => {
  function renderHighlight(fn, props = {}) {
    return fn(<Highlight {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const { container } = renderHighlight(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const html = renderHighlight(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
