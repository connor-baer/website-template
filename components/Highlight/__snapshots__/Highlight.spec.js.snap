// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should render with default styles 1`] = `
.circuit-0 {
  display: inline;
  color: #000;
  background-color: #fff3bf;
}

.circuit-0::before,
.circuit-0::after {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.circuit-0::before {
  content: ' [highlight start] ';
}

.circuit-0::after {
  content: ' [highlight end] ';
}

@media screen and (-ms-high-contrast:active) {
  .circuit-0 {
    color: HighlightText;
    background-color: Highlight;
  }
}

@media print {
  .circuit-0 {
    border: 1pt dotted #000000;
  }
}

<mark
  class="circuit-0 circuit-1"
>
  Bacon ipsum dolor amet swine shank biltong ham bacon.
</mark>
`;
